crates/app/src/builder.rs

```diff
@@ -1,34 +1,62 @@
 // SPDX-FileCopyrightText: 2022 Profian Inc. <opensource@profian.com>
 // SPDX-License-Identifier: AGPL-3.0-only
 
-use super::{handle, RepoStore, TagStore, TreeStore};
+use super::{handle, RepoStoreKey, TagStoreKey, TreeStoreKey};
 
-use std::sync::Arc;
+use drawbridge_store::{New, Storage};
+
+use std::error::Error;
+use std::ffi::OsStr;
+use std::fs::{create_dir, File};
+use std::path::Path;
 
 use axum::handler::Handler;
 use axum::routing::IntoMakeService;
 use axum::{Extension, Router};
+use cap_async_std::fs::Dir;
+
+pub struct Builder<'a> {
+    data_directory: &'a Path,
+}
 
-#[derive(Default)]
-pub struct Builder;
+impl<'a> Default for Builder<'a> {
+    fn default() -> Self {
+        Self {
+            data_directory: Path::new("drawbridge-data"),
+        }
+    }
+}
 
-impl Builder {
+impl<'a> Builder<'a> {
+    /// Constructs a new [Builder].
     pub fn new() -> Self {
-        Self
+        Default::default()
     }
 
-    // TODO: Add configuration functionality
+    /// The directory to be used for all data storage and configuration.
+    pub fn data_directory<S: AsRef<OsStr>>(mut self, path: &'a S) -> Self {
+        self.data_directory = Path::new(path);
+        self
+    }
 
     /// Builds the application and returns Drawbridge instance as a [tower::MakeService].
-    pub fn build(self) -> IntoMakeService<Router> {
-        let repos: Arc<RepoStore> = Default::default();
-        let tags: Arc<TagStore> = Default::default();
-        let trees: Arc<TreeStore> = Default::default();
-        Router::new()
+    pub fn build(self) -> Result<IntoMakeService<Router>, Box<dyn Error>> {
+        let path = Path::new(&self.data_directory);
+
+        if !path.exists() {
+            create_dir(path)?;
+        }
+
+        let root = Dir::from_std_file(File::open(path)?.into());
+        let repo_store = Storage::<RepoStoreKey>::new_arc(root.clone(), "repo".to_string());
```

Comment:
Why use a `Storage` type alias and not a generic type parameter?
If you look in the git history on services, you should be able to find some examples of the bounds.

Reply:
Just being lazy the first time I wrote it. It really doesn't make sense though I can go through and change that.

Reply:
Do you mean remove the feature flag and then have the user decide when they make the builder?

Reply:
I would remove the feature flag and guard the `Memory` by `cfg(test)`, such that it could only be used from within tests explicitly in unit tests